src/main/java/ch/njol/skript/expressions/ExprHotbarSlot.java

```diff
@@ -20,69 +20,117 @@
 
 import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerItemHeldEvent;
 import org.bukkit.inventory.PlayerInventory;
 import org.eclipse.jdt.annotation.Nullable;
 
-import ch.njol.skript.classes.Changer;
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
-import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.skript.util.Getter;
 import ch.njol.skript.util.slot.InventorySlot;
 import ch.njol.skript.util.slot.Slot;
+import ch.njol.util.Kleenean;
 
 @Name("Hotbar Slot")
-@Description({"The slot number of the currently selected hotbar slot."})
-@Examples({"message \"%player's current hotbar slot%\"",
-            "set player's selected hotbar slot to slot 4 of player"})
+@Description({
+	"The slot number of the currently selected hotbar slot.",
+	"Use future and past tense to grab the previous slot in an item change event, see example."
+})
+@Examples({
+	"message \"%player's current hotbar slot%\"",
+	"set player's selected hotbar slot to slot 4 of player",
+	"",
+	"on item held change:",
+		"\tif the selected hotbar slot was a diamond:",
+			"\t\tset the currently selected hotbar slot to slot 5 of player"
+})
 @Since("2.2-dev36")
```

Comment:
```suggestion
@Since("2.2-dev36, INSERT VERSION (Future and past)")
```

Reply:
Not worthy of a tag mention

Reply:
Any signature change should be mentioned to users to at least clear up confusion in that way.